packages/vue-language-service/src/plugins/vue-toggle-v-bind-codeaction.ts

```diff
@@ -0,0 +1,91 @@
+import { LanguageServicePlugin } from '@volar/language-service';
+import { VueFile, walkElementNodes } from '@volar/vue-language-core';
+import { AttributeNode, NodeTypes, RootNode } from 'packages/vue-language-core/src/utils/vue2TemplateCompiler';
+import type * as vscode from 'vscode-languageserver-protocol';
+
+export default function (): LanguageServicePlugin {
+	return (ctx) => {
+
+		return {
+
+			provideCodeActions(document, range, _context) {
+				const startOffset = document.offsetAt(range.start);
+				const endOffset = document.offsetAt(range.end);
+
+				const [vueFile] = ctx!.documents.getVirtualFileByUri(document.uri);
+				if (!vueFile || !(vueFile instanceof VueFile)) {
+					return;
+				}
+
+				const { templateAst, template } = vueFile.sfc;
+
+				if (!templateAst) return;
+
+				const templateStartOffset = template!.startTagEnd;
+				const result: vscode.CodeAction[] = [];
+
+				walkElementNodes(templateAst, node => {
```

Comment:
As I understand `walkElementNodes` would walk literally every single node in AST isn't it bad for performance?

Reply:
Oh, we don't really do any heavy computation here, so nevermind. Will remove findTemplateNode then